### PR TITLE
feat(api): Implement authentication and user management

### PR DESCRIPTION
**Pull request by nfebe**
- Added notification service for email notifications
- Implemented registration, login, and forgot password functionality
- Enhanced User model to support username and phone number for login
- Created generic mailer used by the notification service
- Added user seeder to populate test users
- Created events, listeners, and mailer for user registration
- Updated migrations to include phone and username in the users table
- Created a new migration for verification codes
- Added middleware to ensure JSON request body format

Resolves https://github.com/trakli/trakli/issues/6

### CODE REVIEW

**Review by nfebe — COMMENTED** [review]
I have to fix the openapi doc generation or the composer script something weird is going on with one of them.

File: app/Http/Controllers/API/v1/Auth/ForgotPasswordController.php [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+        $codeEntry = VerificationCode::where('contact', $request->email)
+            ->where('purpose', 'password_reset')
+            ->first();
+
+        if (! $codeEntry || $codeEntry->code != $request->code || $codeEntry->isExpired()) {
+            return $this->failure('Invalid or expired code.', 400);
+        }
+
+        return $this->success([], 'Code verified successfully.');
```
Comment (aanorbel) [inline app/Http/Controllers/API/v1/Auth/ForgotPasswordController.php:142]: Verification code is retrieved but no action is performed on the code to change the status or the user to change verification status.

File: app/Mail/GenericMail.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        //
+    }
+
+    /**
+     * Get the message envelope.
+     */
+    public function envelope(): Envelope
+    {
+        return new Envelope(
+            subject: 'Generic Mail',
```
Comment (aanorbel) [inline app/Mail/GenericMail.php:29]: Keep `i18n` in mind now

File: app/Models/VerificationCode.php [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class VerificationCode extends Model
+{
+    use HasFactory;
+
+    protected $fillable = ['code', 'code', 'purpose', 'expires_at'];
```
Comment (aanorbel) [inline app/Models/VerificationCode.php:12]: `code` appears multiple times

File: tests/Feature/AuthControllerTest.php [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+        $this->assertDatabaseHas('users', ['email' => $response->json('data.user.email')]);
+    }
+
+    public function test_register_validation_error()
+    {
+        $response = $this->postJson('/api/v1/register', []);
+
+        $response->assertStatus(422)
+            ->assertJsonValidationErrors(['email', 'first_name', 'password']);
+    }
```
Comment (aanorbel) [inline tests/Feature/AuthControllerTest.php:44]: Validation with where each rule fails too i neccessary

**Review by aanorbel — CHANGES_REQUESTED** [review]
🚀

File: app/Http/Controllers/API/v1/Auth/AuthController.php [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+    )]
+    public function register(Request $request)
+    {
+        $validatedData = $request->validate([
+            'email' => 'required|string|email|max:255|unique:users',
+            'first_name' => 'required|string|max:255',
+            'last_name' => 'string|max:255',
+            'username' => 'string|max:255',
+            'phone' => 'string|max:255',
+            'password' => 'required|string|min:8',
```
[the remark this answers is not in the document]
Comment (nfebe) [reply app/Http/Controllers/API/v1/Auth/AuthController.php:45]: last_name, username and phone are optional.

[the remark this answers is not in the document]
Comment (aanorbel) [reply app/Http/Controllers/API/v1/Auth/AuthController.php:47]: Was thinking of password constraints. We can change that at any time.

**Review by aanorbel — APPROVED** [review]
🐳  🚀

**Review by aanorbel — APPROVED** [review]

**Pull request merged**